crates/hotshot/task-impls/src/quorum_vote/handlers.rs

```diff
@@ -452,10 +452,18 @@ pub(crate) async fn update_shared_state<
     ) {
         tracing::trace!("{e:?}");
     }
-
+    let update_leaf_duration = now.elapsed();
+
+    consensus_writer
+        .metrics
+        .validate_and_apply_header_duration
+        .add_point(validation_duration.as_secs_f64());
+    consensus_writer
+        .metrics
+        .update_leaf_duration
+        .add_point(update_leaf_duration.as_secs_f64());
```

Comment:
can we get higher resolution duration here, milliseconds would be ok, nanos preferable